hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,244 @@
+        except subprocess.CalledProcessError as e:
+            logger.error(" %s ", e)
+            raise SwitchError('Ovs command failed: ')
+        output = output.split('\n')
+        output.remove('')
+        i_info = dict(s.split(':') for s in output)
+        i_info = {k.strip(): v.strip() for k, v in i_info.iteritems()}
+        # above statement removes extra white spaces from the keys and values.
+        # That is important since other calls will be querying this dictionary.
+        for x in i_info.keys():
+            if i_info[x][0] == "{":
+                i_info[x] = self._string_to_dict(i_info[x])
+            elif i_info[x][0] == "[":
+                i_info[x] = self._string_to_list(i_info[x])
+        return i_info
+
+    def _remove_native_vlan(self, port):
+        """Removes native vlan from a trunked port.
+        If it is the last vlan to be removed, it disables the port and
+        reverts its state to default configuration
+        Args:
+            port: Valid switch port
+        Returns: if successful None else error message
+        """
+        port_info = self._interface_info(port)
+        vlan_id = port_info['tag']
+        shell_cmd = "sudo ovs-vsctl remove port {port} tag {vlan_id}".format(
+                    port=port, vlan_id=vlan_id
+                    )
+        return self.ovs_connect(shell_cmd)
+
+    def _set_native_vlan(self, port, new_network):
+        """Sets native vlan for a trunked port.
+        It enables the port, if it is the first vlan for the port.
+        Args:
+            port: valid port of switch
+            new_network: vlan_id
+        """
+        shell_cmd = (
+                "sudo ovs-vsctl set port {port} tag={netid}".format(
+                        port=port, netid=new_network
+                        ) + " vlan_mode=native-untagged"
+                    )
+
+        return self.ovs_connect(shell_cmd)
+
+    def _add_vlan_to_trunk(self, port, vlan_id):
+        """ Adds vlans to a trunk port. """
+        port_info = self._interface_info(port)
+
+        if not port_info['trunks']:
+            shell_cmd = "sudo ovs-vsctl set port {port} trunks={vlans}".format(
+                    port=port, vlans=vlan_id
+                    )
+        else:
+            all_trunks = ','.join(port_info['trunks'])+','+vlan_id
+            shell_cmd = "sudo ovs-vsctl set port {port} trunks={vlans}".format(
+                        port=port, vlans=all_trunks
+                        )
+
+        return self.ovs_connect(shell_cmd)
+
+    def _remove_vlan_from_port(self, port, vlan_id):
+        """ removes a single vlan specified by `vlan_id` """
+        port_info = self._interface_info(port)
+        shell_cmd = "sudo ovs-vsctl remove port {port} trunks {vlan}".format(
+                      port=port, vlan=vlan_id
+                      )
+        if port_info['trunks']:
+            return self.ovs_connect(shell_cmd)
+        return None
+
+# 3. superclass methods (Not Required):
```

Comment:
Call me nit-picky, but these are definitely required haha.

Reply:
so I will remove the partial comment `(Not Required)`